code/clement/django/lab03/django_pokedex/pokedex_app/views.py

```diff
@@ -0,0 +1,34 @@
+from django.shortcuts import render
+from .models import PokemonType, Pokemon
+from django.db.models import Q
+
+
+# Create your views here.
+def index(request):
+    pokemons = Pokemon.objects.all()
+
+    # Applying various filters base on my form data
+    # And pull values out from the form data
+
+    pokemon_search = request.POST.get('pokemon-search') or ''
+
+    if pokemon_search:
+        # Using Q object to search within the name & types fields for pokemon_search
+        pokemons = Pokemon.objects.filter(
+            Q(name__icontains=pokemon_search) | Q(
```

Comment:
Unlikely to happen with pokemon, but this could cause an issue if you had a pokemon and a Type with the same name, you could never search for the type because it would always search the pokemon instead. Could add a radio box toggle instead.

Reply:
I agree with Mike. A radio button, checkbox or select menu would be nice for choosing to which field the filter query applied. This 'filter-by' value would then be used on the backend to determine how to filter.  

#### Let's just remove the `Q` elements for now and just filter by pokemon name.